pgosupport.cmake

```diff
@@ -42,10 +42,16 @@ function(add_pgo TargetName)
                 set_property(TARGET ${TargetName} APPEND_STRING PROPERTY LINK_FLAGS_RELWITHDEBINFO " /LTCG /USEPROFILE:PGD=${ProfilePath}")
             else(WIN32)
                 if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE OR UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)
-                    if(HAVE_LTO)
-                        target_compile_options(${TargetName} PRIVATE -flto -fprofile-instr-use=${ProfilePath} -Wno-profile-instr-out-of-date)
-                        set_property(TARGET ${TargetName} APPEND_STRING PROPERTY LINK_FLAGS " -flto -fuse-ld=gold -fprofile-instr-use=${ProfilePath}")
-                    endif(HAVE_LTO)
+                    if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.6)
```

Comment:
Good catch on OSX versioning of Clang.

At the moment, we don't produce profile data on/for OSX, so this code path will not execute at all unless on Linux (the `if(EXISTS ${ProfilePath})` check can only possibly return true on Windows and Linux). This will change if we decide to produce packages for OSX in the future, at which point OSX would just treat Clang as >= 3.6 and assume PGO support.

I'd prefer to keep this as-is for now, since the failure case causes a build break later on, and we'll catch it when testing packages with OSX support. (I'm also not aware of plans to add OSX profdata yet.) If we do, we can refactor the logic to consider OS. Does that seem reasonable?